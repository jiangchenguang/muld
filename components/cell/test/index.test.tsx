import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Cell from '..';

describe('Cell', () => {
    afterEach(cleanup);
    it('click event', () => {
        const onClick = jest.fn();
        const { container } = render(<Cell onClick={onClick}>content</Cell>);
        const cell = container.firstChild;
        expect(cell).toHaveClass('mul-cell');
        fireEvent.click(cell!);
        expect(onClick).toHaveBeenCalled();
    });

    it('arrow direction', () => {
        const { asFragment } = render(
            <Cell isLink arrowDirection="down">
                content
            </Cell>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('render slot', () => {
        const { asFragment } = render(
            <Cell icon="custom icon" title="custom title" label="custom label">
                content
            </Cell>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('title-style prop', () => {
        const { asFragment } = render(
            <Cell
                title="custom title"
                titleStyle={{
                    color: 'red',
                }}
            >
                content
            </Cell>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('icon-prefix prop', () => {
        const { asFragment } = render(
            <Cell iconPrefix="my-icon" icon="success">
                content
            </Cell>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
