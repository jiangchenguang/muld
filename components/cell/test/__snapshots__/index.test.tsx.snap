// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell arrow direction 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm kVtctr mul-cell mul-cell--clickable"
    role="button"
    tabindex="0"
  >
    <div
      class="mul-cell__value mul-cell__value--alone"
    >
      content
    </div>
    <i
      class="sc-fznZeY ebCDYP mul-icon mul-icon-arrow-down mul-cell__right-icon"
      content="F007"
      name="arrow-down"
      style="font-size: 20px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Cell icon-prefix prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm kVtctr mul-cell"
  >
    <i
      class="sc-fznZeY cCHXlm my-icon my-icon-success mul-cell__left-icon"
      content="F0C8"
      name="success"
      style="font-size: 20px;"
    />
    <div
      class="mul-cell__value mul-cell__value--alone"
    >
      content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Cell render slot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm kVtctr mul-cell"
  >
    <i
      class="sc-fznZeY dvqeWw mul-icon mul-icon-custom icon mul-cell__left-icon"
      name="custom icon"
      style="font-size: 20px;"
    />
    <div
      class="mul-cell__title"
    >
      <span>
        custom title
      </span>
      <div
        class="mul-cell__label"
      >
        custom label
      </div>
    </div>
    <div
      class="mul-cell__value"
    >
      content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Cell title-style prop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm kVtctr mul-cell"
  >
    <div
      class="mul-cell__title"
      style="color: red;"
    >
      <span>
        custom title
      </span>
    </div>
    <div
      class="mul-cell__value"
    >
      content
    </div>
  </div>
</DocumentFragment>
`;
